openstudiocore/src/model/EnergyManagementSystemMeteredOutputVariable.hpp

```diff
@@ -0,0 +1,182 @@
+
+/** EnergyManagementSystemMeteredOutputVariable is a ModelObject that wraps the OpenStudio IDD object 'OS:EnergyManagementSystem:MeteredOutputVariable'. */
+class MODEL_API EnergyManagementSystemMeteredOutputVariable : public ModelObject {
+ public:
+  /** @name Constructors and Destructors */
+  //@{
+
+  explicit EnergyManagementSystemMeteredOutputVariable(const Model& model, std::string eMSVariableName);
+
+  explicit EnergyManagementSystemMeteredOutputVariable(const Model& model, EnergyManagementSystemActuator& object);
+
+  explicit EnergyManagementSystemMeteredOutputVariable(const Model& model, EnergyManagementSystemSensor& object);
+
+  explicit EnergyManagementSystemMeteredOutputVariable(const Model& model, EnergyManagementSystemGlobalVariable& object);
+
+  explicit EnergyManagementSystemMeteredOutputVariable(const Model& model, EnergyManagementSystemTrendVariable& object);
+
+  explicit EnergyManagementSystemMeteredOutputVariable(const Model& model, EnergyManagementSystemInternalVariable& object);
+
+  explicit EnergyManagementSystemMeteredOutputVariable(const Model& model, EnergyManagementSystemCurveOrTableIndexVariable& object);
+
+  explicit EnergyManagementSystemMeteredOutputVariable(const Model& model, EnergyManagementSystemConstructionIndexVariable& object);
+
+  virtual ~EnergyManagementSystemMeteredOutputVariable() {}
+
+  //@}
+
+  static IddObjectType iddObjectType();
+
+  static std::vector<std::string> updateFrequencyValues();
+
+  static std::vector<std::string> resourceTypeValues();
+
+  static std::vector<std::string> groupTypeValues();
+
+  static std::vector<std::string> endUseCategoryValues();
+
+  /** @name Getters */
+  //@{
+
+  std::string eMSVariableName() const;
+
+  std::string updateFrequency() const;
+
+  boost::optional<std::string> eMSProgramorSubroutineName() const;
+
+  std::string resourceType() const;
+
+  std::string groupType() const;
+
+  std::string endUseCategory() const;
+
+  boost::optional<std::string> endUseSubcategory() const;
+
+  boost::optional<std::string> units() const;
+
+  boost::optional<ModelObject> eMSVariableObject() const;
+
+  boost::optional<EnergyManagementSystemActuator> emsActuator() const;
+  boost::optional<EnergyManagementSystemSensor> emsSensor() const;
+  boost::optional<EnergyManagementSystemGlobalVariable> emsGlobalVariable() const;
+  boost::optional<EnergyManagementSystemTrendVariable> emsTrendVariable() const;
+  boost::optional<EnergyManagementSystemInternalVariable> emsInternalVariable() const;
+  boost::optional<EnergyManagementSystemCurveOrTableIndexVariable> emsCurveOrTableIndexVariable() const;
+  boost::optional<EnergyManagementSystemConstructionIndexVariable> emsConstructionIndexVariable() const;
+
+  //@}
+  /** @name Setters */
+  //@{
+
+  bool setEMSVariableName(const std::string& eMSVariableName);
+
+  bool setUpdateFrequency(const std::string& updateFrequency);
+
+  bool setEMSProgramorSubroutineName(const EnergyManagementSystemProgram& program);
```

Comment:
If you change the getters to have an upper case Or you should change the setters too